libnavigation-core/src/main/java/com/mapbox/navigation/core/fasterroute/FasterRouteController.kt

```diff
@@ -41,6 +41,9 @@ internal class FasterRouteController(
 
     private val fasterRouteRequestCallback = object : RoutesRequestCallback {
         override fun onRoutesReady(routes: List<DirectionsRoute>) {
+            if (directionsSession.routes.isEmpty()) {
```

Comment:
We may not want to update the faster route dialog randomly when the response comes back.

I wonder if we should be moving this to the getStatus poller. That way we'll be able to manage collisions in a reliable way.

Reply:
yeah am working on it :)